samtranslator/model/sam_resources.py

```diff
@@ -802,6 +803,114 @@ def _validate_deployment_preference_and_add_update_policy(
             )
 
 
+class SamCanary(SamResourceMacro):
+    """SAM canary macro."""
+
+    resource_type = "AWS::Serverless::Canary"
+    property_types = {
+        "FunctionName": PropertyType(True, is_str()),
```

Comment:
I would rather not make `FunctionName` a mandatory field and probably use LogicalId in case this one is not provided.